CRM/Report/Form/Member/Detail.php

```diff
@@ -331,7 +331,11 @@ public function from() {
              LEFT JOIN civicrm_membership_payment cmp
                  ON {$this->_aliases['civicrm_membership']}.id = cmp.membership_id
              LEFT JOIN civicrm_contribution {$this->_aliases['civicrm_contribution']}
-                 ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id\n";
+                 ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id
+             JOIN (SELECT MAX(id) id, contact_id  
+                FROM civicrm_contribution  WHERE civicrm_contribution.financial_type_id = 2
```

Comment:
@seamuslee001 I agree the hard coded value isn't a good idea.  I think the query can be reworked to use the civicrm_membership_payment table to verify the contributions are for memberships.  I will update this once I have that working.